### ISSUE TITLE
cannot import name fsopendir

### ISSUE THREAD

**Issue by sgiavasis** [FCP-INDI/C-PAC#627]
Get this import error when trying to create a BIDS sublist in the CPAC sublist generator:

Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/CPAC/GUI/interface/windows/dataconfig_window.py", line 183, in <lambda>
    self.Bind(wx.EVT_BUTTON, lambda event: self.save(event,'run'), id=ID_RUN_EXT)
  File "/usr/local/lib/python2.7/dist-packages/CPAC/GUI/interface/windows/dataconfig_window.py", line 452, in save
    if self.run(path) >0:
  File "/usr/local/lib/python2.7/dist-packages/CPAC/GUI/interface/windows/dataconfig_window.py", line 250, in run
    CPAC.utils.build_sublist.build_sublist(config)
  File "/usr/local/lib/python2.7/dist-packages/CPAC/utils/build_sublist.py", line 919, in build_sublist
    func)
  File "/usr/local/lib/python2.7/dist-packages/CPAC/utils/bids_metadata.py", line 15, in get_metadata_for_nifti
    from fs.opener import fsopendir
ImportError: cannot import name fsopendir

Looking at the API of my current install of fs (2.0.0), there doesn't seem to be any method named fsopendir under fs.opener.

@chrisfilo , is there a change that you've had to adapt for already? I see an opendir method under the FS class in fs.base- not sure if this carries the same functionality. I wanted to check with you first before I apply a fix.

**Comment by egarza** [comment]
I also get a similar error, but only the last import error:

```
saving /media/egarza/INP_MRI_Backup/projects/INP/addimex_tms/data/mri/data_config_subject_list.yaml
Gathering anatomical files...
2017-07-31 17:24:26,436 : Found 19 files!
170731-17:24:26,436 sublist_builder INFO:
	 Found 19 files!
Gathering functional files...
2017-07-31 17:24:26,437 : Found 12 files!
170731-17:24:26,437 sublist_builder INFO:
	 Found 12 files!
2017-07-31 17:24:26,437 : Filtering anatomical files...
170731-17:24:26,437 sublist_builder INFO:
	 Filtering anatomical files...
2017-07-31 17:24:26,437 : Not filtering out any potential sites...
170731-17:24:26,437 sublist_builder INFO:
	 Not filtering out any potential sites...
2017-07-31 17:24:26,437 : Filtered down to 19 files
170731-17:24:26,437 sublist_builder INFO:
	 Filtered down to 19 files
Filtered down to 19 anatomical files
2017-07-31 17:24:26,437 : Not filtering out any potential sites...
170731-17:24:26,437 sublist_builder INFO:
	 Not filtering out any potential sites...
2017-07-31 17:24:26,437 : Filtered down to 12 files
170731-17:24:26,437 sublist_builder INFO:
	 Filtered down to 12 files
Error importing CPAC
cannot import name fsopendir
```
Does this happen only with the BIDS format?

Thanks
Ed

**Comment by chrisfilo** [comment]
Interesting. Since this code was written we have generalized it into a separate package pybids.grabbids - see https://github.com/INCF/pybids/tree/master/bids/grabbids

**Comment by egarza** [comment]
Ok, so I should just install Grabbit and grabbids?
I'm using Linux Ubuntu 16.04 LTS Xenial and the latest CPAC version.

Thanks

Ed

**Comment by ccraddock** [comment]
I don't think that will work. Let Steve and I look over the problem and come up with a plan to fix. Sorry for the hold up. But should be quick.

**Comment by egarza** [comment]
No problem, thanks, Cam. I downloaded the bids/cpac but I cannot load the GUI (getting display error), the reason I want the standalone CPAC for my system is to just run the GUI to create the subject file list.

Thanks again

**Comment by ccraddock** [comment]
Let's debug the GUI issue then. I will switch my comments to that issue.

**Comment by sgiavasis** [comment]
This will be resolved in 1.0.3!
